### PR TITLE
(feat) First cut at a developer's guide to help onboarding users.

### PR DESCRIPTION
**Pull request by jsuereth**
Adds just a few tips & tricks from my own experience working on semconv.   Would like to keep this up-to-date as we finish up policy engine and move beyond just semconv.

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+TODO - Add more getting started guides.
+
+## JQ - Tips and Tricks
+
+While JQ is extremely powerful, it can be finicky to use.  For example, fighting instances where JQ decides to repate a structure vs. nest a list can take some getting used to.
+
+Using an explorer tool like [DevToolsDaily JQ playground](https://www.devtoolsdaily.com/jq_playground/) can dramatically improve your debugging journey in crafting the `jq` expressions you use in rendering templates.
+
+To do so we recommend following [Creating a JSON ouput for your registry](#creating-a-json-output-for-your-registry) and then copy-paste your registry JSON as input into the JQ playground.
```
Comment (lquerel) [inline docs/developer-guide.md]: The method described in #creating-a-json-output-for-your-registry is useful for analyzing the context of a template after JQ processing. However, for the debugging phase of a JQ expression, it is simpler to use the following command: `weaver registry resolve -r <registry>`. For more details on this command `weaver registry resolve --help`.

File: docs/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Templates - Tips and Tricks
+
+When designing JINJA templates, you can make use the the `debug` function to output the context of the tempalte at any point in time.  Simply add `{{ debug(ctx.some_variable) }}` do your template and you'll get a
```
Comment (lquerel) [inline docs/developer-guide.md]:
```suggestion
When designing JINJA templates, you can make use the the `debug` function to output the context of the template at any point in time.  Simply add `{{ debug(ctx.some_variable) }}` to your template and you'll get a
```

File: docs/developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+JSON rendered view of whatever is passed into `debug` at that portion of the template.
+
+## Policies - Tips and tricks
+
+The OPA policy language REGO can be complicated to learn at first.  Using an explorer tool like [The Rego Playground](https://play.openpolicyagent.org) can be a dramatic aide in live debugging policy files and 
+understanding the language.  
+
+To debug `before_resolution` policies, simply take the model YAML files and convert them to JSON (online tools work here) and then copy-paste this data into the "input" of the Rego playground.
+
+To debug `after_resolution` policies, simply follow the [Creating a JSON output for your registry](#creating-a-json-output-for-your-registry) directions and then copy your registry's JSON into the "input" of
```
Comment (lquerel) [inline docs/developer-guide.md:27]: See my previous remark, the weaver registry resolve command will be simpler in this case.

**Review by lquerel — APPROVED** [review]
I made few suggestions.

File: docs/developer-guide.md (hunk 1 above)
Comment (jsuereth) [reply docs/developer-guide.md]: Fixed

File: docs/developer-guide.md (hunk 3 above)
Comment (jsuereth) [reply docs/developer-guide.md:27]: Agreed - I forgot that one was actually implemented, thought it was still TODO :)

File: docs/developer-guide.md (hunk 2 above)
Comment (jsuereth) [reply docs/developer-guide.md]: fixed

**Pull request merged**